File: code/day_027/03_最大子序和/v3_自己错误.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int result=INT_MIN;
        int flag=1;

        if(nums.size()==1) return nums[0];

        for(int i=0; i<nums.size(); i++){

            if(nums[i]>result && flag==1 && result>=0 || result<0) {
                result = nums[i];
                flag = 0;
                continue;
            }
            
            if(result+nums[i] <0){
                flag = 1;
                continue;
            }


            result += nums[i];


            cout << "i:" << i << endl;
            cout << "result:" << result << endl;

        }

        return result;
    }
};

File: leetcode/code/day_002_双指针/03_三数之和/v3_自己错误.py
class Solution:
    def threeSum(self, nums: list[int]) -> list[list[int]]:
        
        result = []

        nums_sorted = sorted(nums)
        for left_1 in range(0, len(nums_sorted)-2):
            right   = len(nums_sorted)-1
            left_2  = left_1+1
        
            while(left_2 < right):
                if nums_sorted[left_1] + nums_sorted[left_2] + nums_sorted[right] == 0:
                    result.append([nums_sorted[left_1], nums_sorted[left_2], nums_sorted[right]])
                    break
                elif nums_sorted[left_1] + nums_sorted[left_2] + nums_sorted[right] < 0:
                    left_2 += 1
                elif nums_sorted[left_1] + nums_sorted[left_2] + nums_sorted[right] > 0:
                    right -= 1

        return result